src/Function.cpp

```diff
@@ -71,11 +71,70 @@ using namespace std;
 
 ///////////////////////////////////////////////////////////////////////////////
 
+static vector<FunctionAttribute*> attributes;
 static vector<Function*> FuncList;		// List of all functions in the program
 static vector<FactMgr*>  FMList;        // list of fact managers for each function
 static long cur_func_idx;				// Index into FuncList that we are currently working on
 static bool param_first=true;			// Flag to track output of commas
 static int builtin_functions_cnt;
+static size_t flag = 0;
```

Comment:
Or I can change `flag` to `attribute_flag`?

Reply:
Please use (and define) a static variable directly in `FunctionAttribute::OutputAttribute`. Make it of the boolean type.

Reply:
> Please use (and define) a static variable directly in `FunctionAttribute::OutputAttribute`. Make it of the boolean type.

It will print `__attribute__` only for 1st function in random program. The static variable needs to be member of `FunctionAttribute` class to print `__attribute__` for all 10 functions in random program.

Reply:
Ok, then I'm fine with how it is now. Just rename the variable to something better and make it a boolean variable.